File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int row = matrix.length;
        int col = matrix[0].length;
        if (matrix[0][0]>target || matrix[row-1][col-1]<target)return false;
        
        for(int i=0;i<row;i++){
            if (matrix[i][col-1]>=target){
                for(int j=0;j<col;j++){
                    if (matrix[i][j]==target) return true;
                }
                break;
            }
        }
        return false;
    }
}

File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        # if target < matrix[0][0] or target > matrix[-1][-1]:
        #     return False
        
        # using BS
        collect = []
        for i in matrix :
            collect.extend(i)
        low = 0
        high = len(collect) - 1
        while low <= high:
            mid = (high+low)//2
            if collect[mid] == target:
                return True
            elif collect[mid] > target:
                high = mid - 1
            elif collect[mid] <target:
                low = mid + 1
        return False
        